src/liquid/utils/cloneAttributes.ts

```diff
@@ -1,28 +1,42 @@
-import { JSXBase } from '@stencil/core/internal'
+export function cloneAttributes(attributesToIgnore: string[] = []) {
+  const attributes = []
 
-export const cloneAttributes = <B extends HTMLElement>(
-  el: B,
-  attributesToIgnore: string[] = []
-) =>
-  Object.values(el.attributes).reduce<JSXBase.HTMLAttributes<B>>(
-    (acc, { name, value }) => {
-      if (
-        !name ||
-        [
-          'style',
-          'id',
-          'class',
-          'slot',
-          'part',
-          ...attributesToIgnore,
-        ].includes(name)
-      ) {
-        return acc
-      }
+  // Clone attributes from props.
+  for (const key in this) {
+    // Component props are getters. Getters don't have a descriptor.
+    // So we can check for component props as follows:
+    if (
+      !Object.getOwnPropertyDescriptor(this, key) &&
+      this[key] !== undefined
+    ) {
+      if (key === 'dirname') console.info(key, this[key])
+      attributes.push({
+        name: key.replaceAll(/([A-Z])/g, '-$1').toLowerCase(),
+        value: this[key],
+      })
+    }
+  }
 
-      acc[name] = value || true
+  // Also clone attributes not defined as props (non-reactive).
+  for (const attr of this.el.attributes) {
+    attributes.push({
+      name: attr.name,
+      value: attr.value,
+    })
+  }
```

Comment:
Seems to me like you don't take care about the `not defined as props`.

```suggestion
  // Also clone attributes not defined as props (non-reactive).
  for (const attr of this.el.attributes) {
    if (!(attr.name in this)) {
      attributes.push({
        name: attr.name,
        value: attr.value,
      })
    }
  }
```